experimental/InjectiveResolutions/src/InjectiveResolutions.jl

```diff
@@ -0,0 +1,1327 @@
+  Q_graded_part::Union{SubquoModule,Nothing}
+
+  function InjMod(A::MonoidAlgebra,I::Vector{IndecInj})
+    return new(A,I,nothing)
+  end
+end
+
+function Q_graded_part(I::InjMod)
+  if I.Q_graded_part === nothing
+    I.Q_graded_part = compute_Q_graded_part(I)
+  end
+  return I.Q_graded_part
+end
+
+function compute_Q_graded_part(I::InjMod)
+  kQ = I.monoid_algebra
+  irreducible_ideals = [_get_irreducible_ideal(kQ, J) for J in I.indec_injectives]
+  irreducible_comp = [quotient_ring_as_module(Ji) for Ji in irreducible_ideals]
+  return direct_sum(irreducible_comp...,task=:none)
+end
+struct IrrRes # irreducible resolution (including all computed data and the cochain complex)
+  mod::SubquoModule
+  irr_sums::Vector{InjMod}
+  cochain_maps::Vector{SubQuoHom}
+  cochain_complex::ComplexOfMorphisms # if sequence not exact return trivial cochain_complex (M0 -> M0)
+end
+
+#direct sum of two injective modules over the same monoid algebra
+function +(I::InjMod, J::InjMod)
+  @req I.monoid_algebra == J.monoid_algebra "monoid algebras not the same"
+  return InjMod(I.monoid_algebra, vcat(I.indec_injectives, J.indec_injectives))
+end
+
+struct InjRes #ZZ^d-graded injective resolution
+  mod::SubquoModule
+  inj_mods::Vector{InjMod}
+  cochain_maps::Vector{MatElem}
+  upto::Int
+  Q_graded_part::IrrRes
+  shift::Vector{Int} #not needed
+end
+
+function Base.show(io::IO,F::FaceQ)
+  print(io,"face corresponding to homogeneous prime ",lowercase(string(F.prime)))
+end
+
+function Base.show(io::IO, ::MIME"text/plain", kQ::MonoidAlgebra)
+  println(io, "monoid algebra over ", lowercase(string(coefficient_ring(kQ))),
+    " with cone of dimension $(dim(cone(kQ)))",
+  )
+  #=
+  println(io,"Monoid algebra k[Q] over ", lowercase(string(coefficient_ring(kQ.algebra))), " generated by ", join(gens(kQ.algebra), ", "), " quotient by ideal (", join(gens(modulus(kQ.algebra)),", "),"), ")
+  print(io, "where Q is a subset of ZZ^", ambient_dim(kQ.cone)," and the cone RR_{>= 0}Q has dimension ", dim(kQ.cone), ".")
+  =#
+end
+
+function Base.show(io::IO, kQ::MonoidAlgebra)
+  print(io, "monoid algebra over ", lowercase(string(coefficient_ring(kQ))),
+    " with cone of dimension $(dim(cone(kQ)))",
+  )
+end
+
+function Base.show(io::IO, ::MIME"text/plain", I::MonoidAlgebraIdeal)
+  println(
+    io, "ideal over $(base_ring(I)) generated by "*join(["$(x)" for x in gens(I)], ", ")
+  )
+end
+
+function Base.show(io::IO,I::MonoidAlgebraIdeal)
+  print(io,underlying_ideal(I))
+end
+
+function Base.show(io::IO,J::InjMod)
+print(
+  io, "injective module given by direct sum of ", length(J.indec_injectives)," indecomposable injectives"
+)
+end
+
+function Base.show(io::IO, ::MIME"text/plain", J::InjMod)
+  println(io, "injective module given by direct sum of indecomposable injectives")
+  for Ji in J.indec_injectives
+    println(io, "  k{", Ji.vector, " + F - Q}, where p_F = ", Ji.face.prime)
+  end
+  print(
+    io,
+    "over ",
+    J.monoid_algebra
```

Comment:
```suggestion
    base_ring(J)
```
...and probably elsewhere?